test/recipes/summerfi-shifter/summerfi-aaveV3-import.js

```diff
@@ -0,0 +1,399 @@
+        enabledAsColl,
+        borrowAddr,
+    } = await view.getLoanData(market, user);
+
+    const collTokenAddresses = collAddr.filter((e) => e !== nullAddress);
+    const useAsCollateralFlags = enabledAsColl.slice(0, collTokenAddresses.length);
+    const debtTokenAddresses = borrowAddr.filter((e) => e !== nullAddress);
+
+    const {
+        collAssetIds,
+        collATokenAddresses,
+    } = await Promise.all(
+        collTokenAddresses.map(async (c) => getAaveReserveData(pool, c)),
+    ).then((arr) => arr.reduce((acc, { id, aTokenAddress }) => ({
+        collAssetIds: [...acc.collAssetIds, id],
+        collATokenAddresses: [...acc.collATokenAddresses, aTokenAddress],
+    }), ({
+        collAssetIds: [],
+        collATokenAddresses: [],
+    })));
+
+    const {
+        debtAssetIds,
+    } = await Promise.all(
+        debtTokenAddresses.map(async (c) => getAaveReserveData(pool, c)),
+    ).then((arr) => arr.reduce((acc, { id }) => ({
+        debtAssetIds: [...acc.debtAssetIds, id],
+    }), ({
+        debtAssetIds: [],
+    })));
+
+    const debtAmounts = await view.getTokenBalances(
+        market,
+        user,
+        debtTokenAddresses,
+    ).then((r) => r.map(({ borrowsVariable }) => borrowsVariable));
+
+    const collAmounts = await view.getTokenBalances(
+        market,
+        user,
+        collTokenAddresses,
+    ).then((r) => r.map(({ balance }) => balance));
+
+    return {
+        collAssetIds,
+        collATokenAddresses,
+        useAsCollateralFlags,
+        collAmounts,
+
+        emodeCategoryId,
+        debtTokenAddresses,
+        debtAssetIds,
+        debtAmounts,
+    };
+};
+
+const validatePositionShift = (oldPosition, newPosition) => {
+    expect(oldPosition.emodeCategoryId).to.be.eq(newPosition.emodeCategoryId);
+    oldPosition.collAssetIds.map((e, i) => expect(e).to.be.eq(newPosition.collAssetIds[i]));
+    oldPosition.collATokenAddresses.map((e, i) => expect(e).to.be.eq(newPosition.collATokenAddresses[i]));
+    oldPosition.useAsCollateralFlags.map((e, i) => expect(e).to.be.eq(newPosition.useAsCollateralFlags[i]));
+    oldPosition.debtTokenAddresses.map((e, i) => expect(e).to.be.eq(newPosition.debtTokenAddresses[i]));
+    oldPosition.debtAssetIds.map((e, i) => expect(e).to.be.eq(newPosition.debtAssetIds[i]));
+
+    oldPosition.collAmounts.map((e, i) => {
+        expect(newPosition.collAmounts[i]).to.be.gte(e);
+        expect(newPosition.collAmounts[i].sub(e)).to.be.lte(e.div(100_00));
+    });
+    oldPosition.debtAmounts.map((e, i) => {
+        expect(newPosition.debtAmounts[i]).to.be.gte(e);
+        expect(newPosition.debtAmounts[i].sub(e)).to.be.lte(e.div(100_00));
+    });
+};
+
+describe('Summerfi-AaveV3-Import', function () {
+    this.timeout(1_000_000);
+    let flAddress;
+
+    const fixture = async () => {
+        await resetForkToBlock(sfProxyLiquidBlock);
+
+        await redeploy('SFApproveTokens');
+        await getContractFromRegistry('AaveV3View');
```

Comment:
Not used

Reply:
It conditionally redeploys if not in registry. AaveV3View is used later on.